dashboard/app/controllers/student_work_sample_controller.rb

```diff
@@ -1,14 +1,68 @@
-class StudentCodeSampleController < ApplicationController
+class StudentWorkSampleController < ApplicationController
   authorize_resource class: false
 
-  # GET /student_code_sample/fetch_student_code_samples
-  def fetch_student_code_samples
-    level_id = student_code_sample_params[:level_id]
-    unit_id = student_code_sample_params[:unit_id]
-    num_samples = student_code_sample_params[:num_samples].to_i
+  # POST /student_work_sample/fetch_free_response_answers
+  def fetch_free_response_answers
+    level_id = student_work_params[:level_id]
+    unit_id = student_work_params[:unit_id]
+    num_samples = student_work_params[:num_samples].to_i
 
     return render json: [] if num_samples == 0
 
+    begin
+      level = Level.find(level_id)
+    rescue ActiveRecord::RecordNotFound
+      return render status: :not_found, json: "Level with id #{level_id}"
+    end
+
+    return render status: :bad_request, json: "Level #{level_id} is not a free response level" unless level.is_a?(FreeResponse)
+
+    begin
+      Unit.find(unit_id)
+    rescue ActiveRecord::RecordNotFound
+      return render status: :not_found, json: "Unit with id #{unit_id}"
+    end
+
+    # We want to pull samples from students who have been assigned to work on the level.
+    sections = Section.where(script_id: unit_id)
+    student_answers = []
+    have_enough_samples = false
+    sections.each do |section|
+      student_ids = Follower.where(section: section).pluck(:student_user_id)
+      student_ids.each do |student_id|
+        unless have_enough_samples
+          student_name = User.find(student_id).username
+          student_answer = get_free_response_answer(student_id, level_id, unit_id)
+          if student_answer
+            student_answers << {
+              level_id: level_id,
+              unit_id: unit_id,
+              student_id: student_id,
+              student_display_name: student_name,
+              student_work: student_answer,
+              section_id: section.id
```

Comment:
I want to include the section id here because we'll want a way to group evaluations for summary analysis and section makes the most sense since that's how they'll be grouped naturally for teachers